### PR TITLE
mark internal API #65

### PR DESCRIPTION
**Pull request by helena**
https://github.com/akka/akka-persistence-cassandra/issues/65

Based on the existing patterns in the repo, the rules I went with were:
* Annotated if a private[akka] top level type, object member case etc, or a def.
* Not annotated if `val`.
* If already having scaladoc or a top-level, added doc comment `INTERNAL API`.

### CODE REVIEW

**Comment by patriknw** [comment]
One primitive but anyway good way of finding public things is to look at the Scaladoc since `private[akka]` doesn't show up there. Things in Scaladoc are public.

I think the actors themselves, such as CassandraJournal can be internal and extending them shouldn't be allowed.

Many other things can also be made internal.

Maybe easiest to talk about what should be public instead, since it shouldn't be much public api.

I think it's mainly:

* CassandraSession
* CassandraReadJournal
* SessionProvider

File: core/src/main/scala/akka/persistence/cassandra/journal/CassandraStatements.scala [hunk 1]
```diff
@@ -4,17 +4,19 @@
...
 import akka.Done
+import akka.annotation.InternalApi
 import akka.cassandra.session.scaladsl.CassandraSession
-import com.datastax.driver.core.Session
-
-import scala.concurrent.{ ExecutionContext, Future }
 import akka.persistence.cassandra.indent
+import com.datastax.driver.core.Session
 
 trait CassandraStatements {
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/cassandra/journal/CassandraStatements.scala]: I think entire CassandraStatements can be internal

File: core/src/main/scala/akka/persistence/cassandra/journal/TagWriter.scala [hunk 2]
```diff
@@ -35,33 +36,36 @@ import scala.concurrent.duration._
  */
 @InternalApi private[akka] object TagWriter {
 
-  private[akka] def props(settings: TagWriterSettings, session: TagWritersSession, tag: Tag): Props =
+  @InternalApi private[akka] def props(settings: TagWriterSettings, session: TagWritersSession, tag: Tag): Props =
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/cassandra/journal/TagWriter.scala]: if the outer `object` is internal it's not necessary to duplicate that for the inner members

Comment (helena) [reply core/src/main/scala/akka/persistence/cassandra/journal/TagWriter.scala]: Ouch, not sure how I missed that.

**Comment by helena** [comment]
Going to close and repost for cleaner PR now https://github.com/akka/akka-persistence-cassandra/pull/596 merged.

**Pull request closed without merging**